### PR TITLE
Add autocapitalize to payee autocomplete

### PR DESCRIPTION
**Pull request by JukeboxRhino**
Having your words be automatically capitalized while typing in payee names is a nice QoL feature that we can add for very little effort on our part. Existing search functionality is maintained because the search is case-insensitive. Only mobile browsers are affected.

Before:
[Screen_Recording_20240714_203146_Firefox.webm](https://github.com/user-attachments/assets/a696c864-bc3e-473d-bf62-5ffde724c307)

After:
[Screen_Recording_20240714_203019_Firefox.webm](https://github.com/user-attachments/assets/46fd422e-83d1-4dcc-84cd-bc6894f0bdf1)

Reference:
[autocapitalize="words"](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/autocapitalize#words)

### CODE REVIEW

**Review by MatissJanis — APPROVED** [review]
This is indeed a very nice QoL addition. Thanks for this!

(will wait for another maintainer to approve before merging)

File: packages/desktop-client/src/components/autocomplete/PayeeAutocomplete.tsx
Comment (MatissJanis) [inline packages/desktop-client/src/components/autocomplete/PayeeAutocomplete.tsx:291]: **💬 suggestion:** ‏I believe the previous lowercase variation was more correct here (according to the spec of this attribute)
```suggestion
        autocapitalize: 'words',
```

Comment (JukeboxRhino) [reply packages/desktop-client/src/components/autocomplete/PayeeAutocomplete.tsx:291]: Will take a look at this later after work, just went with what the autosuggest was wanting so I've got no problems with updating that

Comment (JukeboxRhino) [reply packages/desktop-client/src/components/autocomplete/PayeeAutocomplete.tsx:291]: @MatissJanis the suggestion does not appear to be valid since the type is coming from [here](https://use-form.netlify.app/interfaces/_node_modules__types_react_index_d_.react.inputhtmlattributes#autocapitalize)
image

Comment (MatissJanis) [reply packages/desktop-client/src/components/autocomplete/PayeeAutocomplete.tsx:291]: Thanks for checking! 🙇

**Review by psybers — APPROVED** [review]

**Pull request merged**
